src/signature_detection/ip_signature.py

```diff
@@ -20,13 +20,18 @@ def __init__(self, subnet: str):
 
     def __call__(self, packet: Packet) -> bool:
         if net.IP not in packet:
-            raise Exception
+            run_config.log_event.info('Not a layer 3 packet, not subject to IPSignature inspection.')
```

Comment:
I think we should raise custom exceptions here, and then in our entrypoint file we can catch those exceptions and log their messages in one place rather than having disparate logging calls throughout our code.

Reply:
Which entrypoint file? tiny_hippo_run.py? Raising exceptions is dangerous simply because if they are not caught, the program will crash/halt/stop.

Reply:
Yes the exceptions would be caught in tiny_hippo_run.py, then we could extract the messages that the exceptions return with a traceback and log it from there in one place though. This kind of logging just doesn't seem sustainable to me and should be abstracted if possible.